src/Trakx.YouSign.ApiClient/Utils/ApiKeyCredentialsProvider.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Serilog;
+using Trakx.Utils.Apis;
+using Trakx.Utils.DateTimeHelpers;
+
+namespace Trakx.YouSign.ApiClient.Utils
+{
+    public interface ICopperCredentialsProvider : ICredentialsProvider { };
+    public class ApiKeyCredentialsProvider : ICopperCredentialsProvider, IDisposable
+    {
+        internal const string ApiKeyHeader = "ApiKey";
+        internal const string ApiSignatureHeader = "X-Signature";
+        internal const string ApiTimestampHeader = "X-Timestamp";
+
+        private readonly YouSignApiConfiguration _configuration;
+        private readonly IDateTimeProvider _dateTimeProvider;
+        private readonly CancellationTokenSource _tokenSource;
+
+        private static readonly ILogger Logger = Log.Logger.ForContext<ApiKeyCredentialsProvider>();
+
+        public ApiKeyCredentialsProvider(IOptions<YouSignApiConfiguration> configuration,
```

Comment:
here too, with the correct `Startup.cs` we can inject the `YouSignApiConfiguration` directly rather than wrapping it  in Options (not a big deal), but it makes writing unit tests a bit less verbose

Reply:
Class was fully redesigned.